WordPress/gutenberg
# Allow renaming, duplication and deleting of Navigation menus from Browse Mode Sidebar

## What?


Following the UX suggested in https://github.com/WordPress/gutenberg/issues/50583 this PR implements a new ellipsis menu in browse mode that allows us to rename, duplicate and delete menus

Closes https://github.com/WordPress/gutenberg/issues/50583

## Why?


Some of these actions were not available to the user yet.

~This is part of the effort to improve [Navigation on browse mode](https://github.com/WordPress/gutenberg/issues/50396).~

This PR has subsequently undergone several revisions and so it should be in a good state.

## How?


- Add more menu
- Add actions for rename, duplicate and delete.
- Rename has a modal with local state so that edits can be cancelled on modal dismiss.
- Delete has a confirmatory modal state to avoid accidentally destructive actions.

## Follow Ups

The following items would be good for a followup.

- I think we could improve the loading states for when actions are triggered. For example:
    - not defaulting to displaying `Navigation` at the top of the panel when something is changing
    - show instant feedback when duplicating a menu (currently there is a short delay)
    - possibly optimistically navigating to `/navigation` and then reverting on error.

## Testing Instructions





- Using the new three dot menu, test that you can rename a Navigation menu, duplicate it and delete it.
- Check you can achieve all this with a keyboard only.
- Check that all cancel states work.
- Now repeat the steps above but this time observe the Network tab and block the relevant requests and then retry the action to see the error handling. Note that you can edit the blocked url in dev tools block list in order to substitude IDs .etc.


## Screenshots or screencast 


https://github.com/WordPress/gutenberg/assets/444434/fff062d0-4f67-49a9-9995-8cc31509add4



Co-authored-by: Dave Smith <444434+getdave@users.noreply.github.com>
